### ISSUE TITLE
Confirm various skill interactions

### ISSUE THREAD

**Issue by AlmostMatt** [ajhyndman/fire-emblem-working-title#67]
For some skills I am not 100% sure of how they work. I'll edit this post with assumptions that I make and skill interactions that I am able to verify.
Uncertain/Assumed:
* [ ] Wo Dao does not affect AOE specials
* [x] Lifesteal includes special damage, and is reduced by defensive specials like Aegis and Miracle.
* [ ] If both attacker and defender attack twice (brash assault), and defender has vantage, order is def, atk, atk, def. (This is probably wrong though)
* [ ] pre combat specials apply after hp checks for skills like wary fighter and vantage
* [x] Panic causes buffs to give negative stats, but does not actually create debuffs. It also applies to any other buffs that are added for the duration of panic.
* [ ] Blade tomes ignore debuffs and panic when counting buffs

Confirmed:
* [x] Carrot/Egg Weapons apply 'simultaneously' with Fury
* [x] Pain can trigger when attacked, but only when able to retaliate (a melee unit attacking Azama did not trigger pain but a ranged attack did. a lethal ranged attack did not trigger pain).
* [x] Quick Riposte / Wary Fighter / x-Breaker skills cancel each other (Can't followup and guaranteed followup => depends on speed)
* [x] Tyrfing checks hp percent at the start of combat (also it is an in-combat buff, unlike defiant def)
* [x] Lifesteal excludes overkill damage  (Current implementation in damage-calculator is wrong though)
* [x] 11/21 is not <= 50% and 10/21 is not >= 50% (no rounding for hp conditions)
* [x] Heavy Blade causes your attacks to generate 2 charges instead of 1. When an enemy attacks you the normal 1 charge is generated.
* [x] Guard causes enemy to generate 1 less charge from both attacks and from being attacked.
* [x] Rogue dagger can trigger when attacked, but the def/res buff expires at the start of your turn regardless of when you were buffed
* [x] Buffs expire at the start of your own turn.
* [x] Debuffs expire after performing an action (move/atk).
* [x] Postcombat passives trigger if the hero survives, regardless of whether or not they were able to attack
* [x] Postcombat weapons trigger if the hero hit something, regardless of whether or not they survived.
  -- tested with Seal X, Rogue Dagger, Deathly Dagger, Poison Strike, Pain, Slow, Panic, Ragnarok
* [x] The HP condition on Ragnarok applies to both the atk/spd and the post-combat dmg. Ragnarok also cares about attacking, not about initiating.

**Comment by AlmostMatt** [comment]
Also: do pre combat specials apply before hp checks for skills like wary fighter and vantage?

**Comment by ajhyndman** [comment]
The very last comment on this thread supports the conclusion that quick riposte/breaker skills cancel each other out:

https://www.gamefaqs.com/boards/204447-fire-emblem-heroes/75098215?page=2

**Comment by ajhyndman** [comment]
Oh, I guess that was already in your confirmed list.

**Comment by AlmostMatt** [comment]
Ok so Guard and Heavy Blade have really bad descriptions.

This is what they seem to actually do:
Heavy Blade:  your attacks reduce your special cooldown by 2, when someone attacks you your cooldown is reduced by 1.

Guard: your opponent's cooldown remains constant no matter how many times they attack you or how many times you attack them. However if the special is already ready they can use it.

**Comment by ajhyndman** [comment]
Yeah, I noticed that these descriptions are the first time the term **cooldown charge** has been used.  It seems to be an inverse concept to **cooldown count**.

**Comment by AlmostMatt** [comment]
Apparently my assumption about lifesteal and overkill was incorrect. I did 10 damage to a 3 health enemy with 30% lifesteal and gained 0 life, so it probably only counts the actual amount of health lost. I'll try to do a few more tests when I find some time.

**Comment by AlmostMatt** [comment]
Testing hp percentage conditions.
36/41 is not considered >= 90%  (tested with Wary fighter 1)
9/17 is not considered <= 50% (tested with Wings of Mercy 3)
So it seems like rounding is not relevant for <= and >= conditions.

**Comment by AlmostMatt** [comment]
Another question:  What is the attack order if both players attack twice and the defender has vantage?
(For example, a fast unit vs Vantage Hector, or a Brash assault unit vs a fast vantage unit)

Edit: Actually with the Hector example, hp has to be >80% for armads and <75% for  vantage.
The brash assault example would still work, if both units are low, but it would be very rare.

**Comment by ajhyndman** [comment]
I'm fairly confident that the answer is Defender > Attacker > Defender > Attacker.  I do not have evidence to back this up, however.

Mass Duel Simulator agrees with this resolution.

**Comment by AlmostMatt** [comment]
I ran into Seliph in the Arena and was able to confirm that Tyrfing does not apply to the second hit if the first hit put him below 50% hp => hp is checked at beginning of combat.

**Comment by ajhyndman** [comment]
By the way, you can make checklists in github comments with the following syntax:

```md
* [ ] incomplete item
* [x] complete item
```

* [ ] incomplete item
* [x] complete item

**Comment by AlmostMatt** [comment]
Apparently Ragnarok is another mistranslated / poorly worded skill.

Tooltip: "If unit has 100% HP at the start of combat, unit receives Atk/Spd +5. If attacking, unit will receive 5 damage after combat."

Actual effect: "If unit has 100% HP at the start of combat, unit receives Atk/Spd +5. If unit attacks or is able to counterattack and is at 100% HP, unit will receive 5 damage after combat."
